site/zenodo_rdm/legacy/generators.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo legacy permissions generators."""
+
+from invenio_rdm_records.services.generators import IfRestricted
+
+
+class IfFilesRestrictedForCommunity(IfRestricted):
+    """Conditional generator for files restriction for community."""
+
+    def __init__(self, then_, else_):
+        """Constructor."""
+        super().__init__("files", then_, else_)
+
+    def _condition(self, record, **kwargs):
+        """Check if community can access restricted files of the migrated record."""
+        # can_community_read_files = getattr(record.parent, permission_flags.can_community_read_files, True)
+        can_community_read_files = False
```

Comment:
Leftover?
```suggestion
        can_community_read_files = getattr(record.parent, permission_flags.can_community_read_files, True)
```

Reply:
Not the leftover, will be uncommented and tested after that field is added: https://github.com/inveniosoftware/invenio-rdm-records/pull/1290